### PR TITLE
Rewrite how ClientFX file extensions are checked.

### PR DESCRIPTION
**Pull request by rohit-n**
This removes a Valgrind invalid read error as well as a platform #ifdef.

### CODE REVIEW

File: NOLF2/ClientShellDLL/ClientShellShared/ClientFXDB.cpp [hunk 1]
```diff
@@ -142,6 +142,7 @@ CClientFXDB::~CClientFXDB()
 
 bool CClientFXDB::Init(ILTClient* pLTClient)
 {
+	char* ext;
 	// Try and load our ClientFX dll (.fxd)
```
Comment (Katana-Steel) [inline NOLF2/ClientShellDLL/ClientShellShared/ClientFXDB.cpp]: Please just declare ext at point of use.

File: NOLF2/ClientShellDLL/ClientShellShared/ClientFXDB.cpp [hunk 2]
```diff
@@ -160,25 +161,16 @@ bool CClientFXDB::Init(ILTClient* pLTClient)
...
-			_splitpath( pEntry->m_pFullFilename, drive, dir, fname, ext );
-
-			// Is this a ClientFx file?
-			if( !stricmp( ".fxf", ext ) )
-#endif
+			if (strlen(pEntry->m_pFullFilename) > 4)
 			{
-				// Try and load it
-				if( !LoadFxGroups( pLTClient, pEntry->m_pFullFilename ) )
+				ext = pEntry->m_pFullFilename + strlen(pEntry->m_pFullFilename) - 4;
```
Comment (Katana-Steel) [inline NOLF2/ClientShellDLL/ClientShellShared/ClientFXDB.cpp]: So `const char *ext = ...` here

File: NOLF2/ClientShellDLL/ClientShellShared/ClientFXDB.cpp [hunk 3]
```diff
@@ -160,25 +161,16 @@ bool CClientFXDB::Init(ILTClient* pLTClient)
...
-			char dir[_MAX_DIR];
-			char fname[_MAX_FNAME];
-			char ext[_MAX_EXT];
-
-			_splitpath( pEntry->m_pFullFilename, drive, dir, fname, ext );
-
-			// Is this a ClientFx file?
-			if( !stricmp( ".fxf", ext ) )
-#endif
+			if (strlen(pEntry->m_pFullFilename) > 4)
```
Comment (Katana-Steel) [inline NOLF2/ClientShellDLL/ClientShellShared/ClientFXDB.cpp]: Optionally do this  along with the if above, like:
```
if(pEntry->m_Type == TYPE_FILE && strlen(pEntry->m_pFullFilename) > 4 )

**Comment by rohit-n** [comment]
Changes done.

**Comment by Katana-Steel** [comment]
Looks good, thank you

**Pull request merged**